### PR TITLE
Merge 1.0.2 setuid calls to getenv(3) safety.

### PR DESCRIPTION
**Pull request by paulidale**
Manual merge of #7047 to 1.0.2-stable.

### CODE REVIEW

**Comment by paulidale** [comment]
I think some of the AppVeyour failures are relevant.  I suspect I'm doubly including a header file.

**Comment by bernd-edlinger** [comment]
ah, yes:
```
randfile.c: In function 'RAND_file_name':
randfile.c:330:9: error: implicit declaration of function 'ossl_safe_getenv' [-Werror=implicit-function-declaration]
     s = ossl_safe_getenv("RANDFILE");
         ^
randfile.c:330:7: error: assignment makes pointer from integer without a cast [-Werror=int-conversion]
     s = ossl_safe_getenv("RANDFILE");
       ^
randfile.c:335:11: error: assignment makes pointer from integer without a cast [-Werror=int-conversion]
         s = ossl_safe_getenv("HOME");
```

**Comment by paulidale** [comment]
I thought I'd fixed that one earlier today.....

**Comment by bernd-edlinger** [comment]
never mind.
I used the OpenSSL_1_0_2-stable branch from github.

**Comment by bernd-edlinger** [comment]
Unless we fix the OpenSSL_1_0_2-stable branch the CI won't be able to produce reasonable results.

**Comment by bernd-edlinger** [comment]
We could force-push the previous revision
```
git checkout OpenSSL_1_0_2-stable
git pull
git reset --hard HEAD^
git push -f
```

**Comment by mspncp** [comment]
Wouldn't it  be better to push a reverting commit instead of force pushing?

**Comment by paulidale** [comment]
Reordering the #include lines seems to have fixed the immediate AppVeyor failures.

**Comment by bernd-edlinger** [comment]
Yes, however the ordering of include files should not have that effect, see #7303

**Comment by paulidale** [comment]
There is a known including file ordering issue with `e_os.h` (it must be first) but I agree, we shouldn't have such dependencies.

File: crypto/conf/conf_api.c [hunk 1]
```diff
@@ -63,6 +63,7 @@
 # define NDEBUG
 #endif
 
+#include "cryptlib.h"
```
Comment (bernd-edlinger) [inline crypto/conf/conf_api.c]: most files include all system headers (stdlib.h, string.h etc.) before "cryptlib.h"

Comment (paulidale) [reply crypto/conf/conf_api.c]: Shuffled it down...

**Comment by bernd-edlinger** [comment]
Re-confirmed.

**Comment by paulidale** [comment]
Merged, thanks.

**Pull request closed without merging**